Makefile

```diff
@@ -0,0 +1,11 @@
+.PHONY: list
+
+list:
+	@LC_ALL=C $(MAKE) -pRrq -f $(lastword $(MAKEFILE_LIST)) : 2>/dev/null | awk -v RS= -F: '/^# File/,/^# Finished Make data base/ {if ($$1 !~ "^[#.]") {print $$1}}' | sort | egrep -v -e '^[^[:alnum:]]' -e '^$@$$'
```

Comment:
This line deserves a comment...

Also, if we use a makefile, I think it should implement the `all` and `clean` conventions